src/app/[entry_id]/page.jsx

```diff
@@ -0,0 +1,100 @@
+"use client"
+
+import React from 'react'
+import { useEffect, useState } from 'react'
+import { supabase } from '../lib/supabase'
+import Link from 'next/link'
+import Image from 'next/image'
+
+function Entry({ params }) {
+  const [entryData, setEntryData] = useState();
+  const entry_id = params.entry_id
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        let { data: diary_entry, error } = await supabase
+          .from('diary_entries')
+          .select()
+          .eq('id', entry_id)
+        if (error) {
+          throw error;
+        }
+        setEntryData(diary_entry[0])
+      } catch (error) {
+        console.error('Error fetching data from Supabase:', error.message);
+      }
+    };
+    fetchData();
+  }, []);
+
+  return (
+    <>
+      {entryData &&
+        <>
+          <div>
+            <div className="px-4 sm:px-0">
+              <h3 className="text-base font-semibold leading-7 text-gray-900">
+                {entryData.date && new Date(entryData.date).toLocaleDateString()}
+              </h3>
+              <p className="mt-1 max-w-2xl text-sm leading-6 text-gray-500">{`ID: ${entryData.id}`}</p>
+            </div>
+            <div className="mt-6 border-t border-gray-100">
+              <dl className="divide-y divide-gray-100">
```

Comment:
you have my respect here, not often you see the `dl` tag. I constantly forget to use it...